src/internal/api/packages/read.go

```diff
@@ -36,7 +37,7 @@ func readPackage(w http.ResponseWriter, path string) (pkg types.APIZarfPackage,
 
 	tmpDir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
 	if err != nil {
-		return pkg, err
+		return pkg, fmt.Errorf("tmpdir does not exist: %w", err)
```

Comment:
This would be better stated as "unable to create tmpdir: ".  There are other reasons this may fail (permissions, bad characters, etc)